text/0052-tx-summary-digest.md

```diff
@@ -0,0 +1,510 @@
+  in order to efficiently compute the (new) digest which MLSAG's must sign.
+* Small additional data (`TxSummaryUnblindingData`) can be streamed with the pieces of the `TxSummary`, allowing
+  the device to "unblind" the `TxSummary`.
+* This can be used to create a `TxSummaryUnblindingReport`, which deduces the following:
+  * For each output, the amount, and the entity to which it is destined, which is one of three possibilities:
+    * The signer themselves
+    * A known contact, identified by their public address
+    * An anonymous swap counterparty
+  * For each input, the amount, and the entity from whcih it came, which is one of two possibilities:
+    * The signer themselves
+    * An anonymous swap counterparty
+
+The design goals here are:
+* A hardware device which is expected to sign MLSAGs for a transaction, can do so while having **visibility
+  into how much funds** are being moved **from whom**, and **to whom** they are going.
+* A hardware device can use the `TxSummaryUnblindingReport` to display this (verified) information to the user.
+* The hardware device need not trust the host computer to verify this information.
+
+There is exactly one possible `TxSummary` for a given `Tx`.
+
+To achieve the goals, and make it unnecessary for the device to see the majority of the `Tx`, we have
+to change the way that the digest which `RingMLSAG`'s sign is computed.
+
+**Note**: This change does not impact the digest that [MCIP 31](0031-transactions-with-contingent-inputs.md) signed contingent inputs (SCIs) sign,
+those inputs sign the same digest that was specified in MCIP 31.
+
+During transaction construction, the signer:
+1. Computes the `extended-message-digest` as before
+1. Creates a merlin transcript, appending that digest (using the `mc-crypto-digestible-scheme`)
+1. Appends `TxSummary` (using the `mc-crypto-digestible` scheme)
+1. Extracts 32 bytes from this digest, the `extended-message-and-tx-summary-digest`.
+
+The MLSAG's (except for SCIs) will now sign this digest (instead of the `extended-message-digest` as previously).
+
+The transaction verifier similarly computes this
+`extended-message-and-tx-summary digest` and verifies that the MLSAGs sign this.
+
+**Note**: This means that the `TxSummary` needs to be constructible by the consensus enclave, given only
+the `Tx`, and therefore cannot contain any of the transaction secrets needed to unblind the `TxSummary`. These
+secrets appear in `TxSummaryUnblindingData` instead. The host computer (ultimately the `TransactionBuilder`)
+is responsible to compute this when it creates
+a `Tx`, and then it can be streamed to, and verified by, the hardware device which is asked to sign.
+
+When a hardware wallet is asked to sign an MLSAG, we can give it now the `extended_message_digest`
+and steam it the `TxSummary` and the `TxSummaryUnblindingData`, and it can compute the appropriate digest from this for the MLSAG to sign.
```

Comment:
```suggestion
and stream it the `TxSummary` and the `TxSummaryUnblindingData`, and it can compute the appropriate digest from this for the MLSAG to sign.
```